crates/db_views/post/src/impls.rs

```diff
@@ -234,12 +244,30 @@ impl PostView {
   }
 }
 
+#[derive(Clone)]
+pub enum CommunityOrMulti {
```

Comment:
See above, its not necessary.
just add `multi_community_ids: Option<Vec<CommunityId>>` as a field to postquery, and use `.eq_any`